src/org/traccar/database/ConnectionManager.java

```diff
@@ -122,6 +124,7 @@ public void updateDevice(final long deviceId, String status, Date time) {
                 public void run(Timeout timeout) throws Exception {
                     if (!timeout.isCancelled()) {
                         updateDevice(deviceId, Device.STATUS_UNKNOWN, null);
+                        activeDevices.remove(deviceId);
```

Comment:
Assuming device is not active when its state unknown, to not send commands in emptiness.